demos/mqtt/mqtt_demo_mutual_auth/mqtt_demo_mutual_auth.c

```diff
@@ -72,16 +83,34 @@
 #ifndef ROOT_CA_CERT_PATH
     #error "Please define path to Root CA certificate of the MQTT broker(ROOT_CA_CERT_PATH) in demo_config.h."
 #endif
-#ifndef CLIENT_CERT_PATH
-    #error "Please define path to client certificate(CLIENT_CERT_PATH) in demo_config.h."
-#endif
-#ifndef CLIENT_PRIVATE_KEY_PATH
-    #error "Please define path to client private key(CLIENT_PRIVATE_KEY_PATH) in demo_config.h."
-#endif
 #ifndef CLIENT_IDENTIFIER
     #error "Please define a unique CLIENT_IDENTIFIER."
 #endif
 
+/* AWS IoT message broker would require either a set of client certificate/private key
+ * or username/password to authenticate the client. */
+#ifndef CLIENT_USERNAME
+    #ifndef CLIENT_CERT_PATH
+        #error "Please define path to client certificate(CLIENT_CERT_PATH) in demo_config.h."
+    #endif
+    #ifndef CLIENT_PRIVATE_KEY_PATH
+        #error "Please define path to client private key(CLIENT_PRIVATE_KEY_PATH) in demo_config.h."
+    #endif
+#else
+
+/* If username is defined, a client password also would need to be defined for
+ * client authentication. */
+    #ifndef CLIENT_PASSWORD
+        #error "Please define client password(CLIENT_PASSWORD) in demo_config.h for client authentication based on username/password."
+    #endif
+
+/* AWS IoT MQTT broker port needs to be 443 for client authentication based on
+ * username/password. */
+    #if AWS_MQTT_PORT != 443
+        #error "Please define broker port(AWS_MQTT_PORT) as 443 in demo_config.h for client authentication based on username/password."
```

Comment:
Could be mentioned as "_Broker port, AWS_MQTT_PORT, should be defined_ as 443 in demo_config.h for client authentication based on username/password"

Reply:
Agree. Sounds good. Thanks